src/main/java/com/fasterxml/jackson/core/filter/FilteringParserDelegate.java

```diff
@@ -418,7 +418,7 @@ public JsonToken nextToken() throws IOException
                 }
                 _itemFilter = f;
                 if (f == TokenFilter.INCLUDE_ALL) {
-                    if (_includePath) {
+                    if (_verifyAllowedMatches() && _includePath) {
```

Comment:
Sure. I've found it at `testNotAllowMultipleMatchesWithoutPath1` test. It's easier to explain based on the line `FilterParserDelegate`:586 - the condition is the same.
If the first match occurs, `_matchCount` has to be incremented and regardless if the `_includePath` is true or false - the rest of the logic will handle both. However, if we invert it - logic will do the job - the output will be ok, but `_matchCount` will stay at 0.

In this commit, `FilterParserDelegate`:421 it's just a copy of the other condition, but I've found it crucial for another PR that I'll publish after this one (more edge cases handled, more tests, one thing at a time)